test/common/alerts_exist.go

```diff
@@ -0,0 +1,281 @@
+		"UnifiedPushDown",
+		"UnifiedPushJavaDeadlockedThreads",
+		"UnifiedPushJavaGCTimePerMinuteScavenge",
+		"UnifiedPushJavaHeapThresholdExceeded",
+		"UnifiedPushJavaNonHeapThresholdExceeded",
+		"UnifiedPushMessagesFailures",
+	},
+	"/redhat-rhmi-rhsso": {
+		"KeycloakAPIRequestDuration90PercThresholdExceeded",
+		"KeycloakAPIRequestDuration99.5PercThresholdExceeded",
+		"KeycloakInstanceNotAvailable",
+		"KeycloakJavaDeadlockedThreads",
+		"KeycloakJavaGCTimePerMinuteMarkSweep",
+		"KeycloakJavaGCTimePerMinuteScavenge",
+		"KeycloakJavaHeapThresholdExceeded",
+		"KeycloakJavaNonHeapThresholdExceeded",
+		"KeycloakLoginFailedThresholdExceeded",
+	},
+	"/redhat-rhmi-user-sso": {
+		"KeycloakAPIRequestDuration90PercThresholdExceeded",
+		"KeycloakAPIRequestDuration99.5PercThresholdExceeded",
+		"KeycloakInstanceNotAvailable",
+		"KeycloakJavaDeadlockedThreads",
+		"KeycloakJavaGCTimePerMinuteMarkSweep",
+		"KeycloakJavaGCTimePerMinuteScavenge",
+		"KeycloakJavaHeapThresholdExceeded",
+		"KeycloakJavaNonHeapThresholdExceeded",
+		"KeycloakLoginFailedThresholdExceeded",
+	},
+}
+
+var expectedExtRules = map[string][]string{
+	"/redhat-rhmi-3scale": {
+		"ThreeScalePostgresUnavailable",
+		"ThreeScalePostgresConnectivity",
+		"BackendRedisUnavailable",
+		"BackendRedisConnectivity",
+		"SystemRedisUnavailable",
+		"SystemRedisConnectivity",
+	},
+	"/redhat-rhmi-codeready": {
+		"CodeReadyPostgresUnavailable",
+		"CodeReadyPostgresConnectivity",
+	},
+	"/redhat-rhmi-ups": {
+		"UPSPostgresUnavailable",
+		"UPSPostgresConnectivity",
+	},
+	"/redhat-rhmi-user-sso": {
+		"UserSSOPostgresUnavailable",
+		"UserSSOPostgresConnectivity",
+	},
+	"/redhat-rhmi-sso": {
+		"SSOPostgresUnavailable",
+		"SSOPostgresConnectivity",
+	},
+	"/redhat-rhmi-amq-online": {
+		"AMQOnlineSSOPostgresUnavailable",
+		"AMQOnlineSSOPostgresConnectivity",
+	},
+}
+
+func TestIntegreatlyAlertsExist(t *testing.T, ctx *TestingContext) {
+	// get the RHMI custom resource to check what storage type is being used
+	rhmi := &v1alpha1.RHMI{}
+	ns := fmt.Sprintf("%soperator", namespacePrefix)
+	err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: installationName, Namespace: ns}, rhmi)
+	if err != nil {
+		t.Fatal("error getting RHMI CR:", err)
+	}
+
+	// add external database alerts to list of expected rules if
+	// is not being used
+	if !rhmi.Spec.UseClusterStorage {
+		for file, rules := range expectedExtRules {
+			expectedRules[file] = append(expectedRules[file], rules...)
+		}
+	}
+
+	// exec into the prometheus pod
+	output, err := execToPod("curl localhost:9090/api/v1/rules",
+		"prometheus-application-monitoring-0",
+		namespacePrefix+"middleware-monitoring-operator",
+		"prometheus", ctx)
+	if err != nil {
+		t.Fatal("failed to exec to pod:", err)
+	}
+
+	// get all found rules from the prometheus api
+	var promApiCallOutput apiResponse
+	err = json.Unmarshal([]byte(output), &promApiCallOutput)
+	if err != nil {
+		t.Logf("Failed to unmarshall json: %s", err)
```

Comment:
Will this allow the test to go past L183 if it cannot unmarshal the response, is this what we want?